src/main/java/clzzz/helper/logic/parser/slot/SlotParserUtil.java

```diff
@@ -79,6 +80,15 @@ public static LocalDateTime parseDateTime(String dateTime) throws ParseException
         } catch (DateTimeParseException e) {
             throw new ParseException(MESSAGE_INVALID_DATETIME);
         }
+        if (!DateTimeUtil.isValidDateString(trimmedDateTime.split("\\s+")[0])) {
+            if (!DateTimeUtil.isValidTimeString(trimmedDateTime.split("\\s+")[1])) {
+                throw new ParseException(Messages.MESSAGE_INVALID_DATE_TIME);
+            }
+            throw new ParseException(Messages.MESSAGE_INVALID_DATE);
+        }
+        if (!DateTimeUtil.isValidTimeString(trimmedDateTime.split("\\s+")[1])) {
+            throw new ParseException(Messages.MESSAGE_INVALID_TIME);
+        }
```

Comment:
This is a library class, so let's not be stingy.
LocalDate.parse() -> you can check invalid date
LocalTime.parse() -> you can check invalid time
LocalDateTime.of(LocalDate date, LocalTime time)